src/lib.rs

```diff
@@ -221,7 +221,7 @@ struct sock_fprog {
 }
 
 /// Library Result type.
-pub type Result<T> = std::result::Result<T, Error>;
+pub type Result<T, E = Error> = std::result::Result<T, E>;
```

Comment:
I don't understand why this is desirable. This is a result specific to seccompiler, not a general-purpose result type alias.

This is a common pattern in rust library crates, for example: https://docs.rs/serde_json/latest/serde_json/type.Result.html

Reply:
> I don't understand why this is desirable. 

This is just a simple QoL improvement. Many libraries have switched to making the error type overridable, because it is an improvement with no downsides.

Reply:
I see your point but I'd rather not have it in seccompiler.

Honestly I'd find it a bit confusing in an application to see code depending on the seccompiler `Result` type that is doing something completely different than seccomp. I also wouldn't want to break a ton of other code if we ever remove this Result type